build.gradle

```diff
@@ -1,29 +1,44 @@
 import org.ajoberstar.grgit.Grgit
-
-buildscript {
-  repositories {
-    google()
-    mavenCentral()
-  }
-  dependencies {
-    classpath libs.grgit.plugin
-    classpath libs.android.gradle.plugin
-    classpath libs.kotlin.gradle.plugin
-    classpath libs.gradle.maven.publish.plugin
-    classpath libs.dokka.gradle.plugin
-  }
-}
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 plugins {
   id 'net.ltgt.errorprone' apply false
   id "com.google.devtools.ksp" apply false
+  id "org.ajoberstar.grgit" apply false
+  id "com.android.library" apply false
+  id "org.jetbrains.kotlin.jvm" apply false
+  id "com.vanniktech.maven.publish" apply false
+  id "org.jetbrains.dokka" apply false
 }
 
 allprojects {
-  repositories {
-    google()
-    mavenCentral()
-    maven { url "https://jitpack.io" }
+  def compileJavaVersion = JavaVersion.VERSION_17
+  def targetJavaVersion = JavaVersion.VERSION_1_8
```

Comment:
This is what I meant as centralized place to define Java compatibility. 
Right now the project compiles with Java 17 (via [toolchains](https://docs.gradle.org/current/userguide/toolchains.html)), but produces artifacts compatible down to `Java 8`